File: src/yanzm/example/adaptersamples/ArrayAdapterSample.java
package yanzm.example.adaptersamples;

import android.app.ListActivity;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.ArrayAdapter;

public class ArrayAdapterSample extends ListActivity {

	public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ArrayAdapter<String> arrayAdapter = new ArrayAdapter<String>(this, R.layout.list_item3);
        setListAdapter(arrayAdapter);
    }

	@Override
	public boolean onCreateOptionsMenu(Menu menu) {
		menu.add(0, 0, 0, "Add");
		menu.add(0, 1, 0, "clear");
		return true;
	}

	@SuppressWarnings("unchecked")
	@Override
	public boolean onOptionsItemSelected(MenuItem item) {
		switch (item.getItemId()) {
		case 0:
			((ArrayAdapter<String>)getListAdapter()).add("New Item");
			return true;
		case 1:
			((ArrayAdapter<String>)getListAdapter()).clear();			
			return true;
		}
		return false;
	}
}

File: app/src/main/java/yanzm/example/adaptersamples/ArrayAdapterSample.kt
package yanzm.example.adaptersamples

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.ArrayAdapter
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity

class ArrayAdapterSample : AppCompatActivity() {

    private lateinit var arrayAdapter: ArrayAdapter<String>

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val listView = ListView(this)
        setContentView(listView)

        arrayAdapter = ArrayAdapter<String>(this, R.layout.list_item3)
        listView.adapter = arrayAdapter
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menu.add(0, 0, 0, "Add")
        menu.add(0, 1, 0, "clear")
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            0 -> {
                arrayAdapter.add("New Item")
                true
            }
            1 -> {
                arrayAdapter.clear()
                true
            }
            else -> {
                super.onOptionsItemSelected(item)
            }
        }
    }
}
